src/en/mangadistrict/src/eu/kanade/tachiyomi/extension/en/mangadistrict/MangaDistrict.kt

```diff
@@ -1,11 +1,124 @@
 
+import android.app.Application
+import android.content.SharedPreferences
+import androidx.preference.ListPreference
+import androidx.preference.PreferenceScreen
+import androidx.preference.SwitchPreferenceCompat
 import eu.kanade.tachiyomi.multisrc.madara.Madara
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.util.asJsoup
+import okhttp3.MediaType
+import okhttp3.Response
+import okhttp3.ResponseBody
+import okhttp3.ResponseBody.Companion.toResponseBody
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
 
 class MangaDistrict : Madara(
     "Manga District",
     "https://mangadistrict.com",
     "en",
 ) {
-    override fun searchMangaNextPageSelector() = "div[role=navigation] a.last"
+
+    private val preferences: SharedPreferences by lazy {
+        Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
+    }
+
+    override fun searchMangaNextPageSelector() = "div[role=navigation] span.current + a.page"
+
+    private val titleComment = Regex("\\(.*\\)")
+
+    override fun popularMangaFromElement(element: Element): SManga {
+        return super.popularMangaFromElement(element).apply {
+            if (preferences.getBoolean(REMOVE_TITLE_COMMENT_PREF, true)) {
+                title = this.title.replace(titleComment, "").trim()
+            }
+        }
+    }
+
+    override fun latestUpdatesFromElement(element: Element) = popularMangaFromElement(element)
+
+    override fun searchMangaFromElement(element: Element): SManga {
+        return super.searchMangaFromElement(element).apply {
+            if (preferences.getBoolean(REMOVE_TITLE_COMMENT_PREF, true)) {
+                title = this.title.replace(titleComment, "").trim()
+            }
+        }
+    }
+
+    override fun mangaDetailsParse(document: Document): SManga {
+        return super.mangaDetailsParse(document).apply {
+            if (preferences.getBoolean(REMOVE_TITLE_COMMENT_PREF, true)) {
+                title = this.title.replace(titleComment, "").trim()
+            }
+        }
+    }
+
+    private fun chaptersStripSelector(): String {
+        return when (getImgResPref()) {
+            "Full quality" -> "a:contains(High Quality) ~ ul.list-chap"
+            "High quality" -> "a:contains(Full Quality) ~ ul.list-chap"
+            else -> "dummy"
+        }
+    }
+
+    private fun Response.stripElements(selector: String): Response {
+        val contentType: MediaType? = body.contentType()
+        val document = asJsoup()
+        document.select(selector).remove()
+
+        val body: ResponseBody = document.body().toString().toResponseBody(contentType)
+        return newBuilder().body(body).build()
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        return super.chapterListParse(response.stripElements(chaptersStripSelector()))
+    }
```

Comment:
don't rebuild the response, just reimplement the chapterListParse.

Also, wouldn't this remove chapters if only "High quality" or "low quality" chapter exists?

Reply:
That's why I don't select the required quality, but I remove the oposite quality instead. If manga doesn't have separate qualities then the selector won't exist and nothing will be removed.
Would you explain why shouldn't rebuild the response? I want to call the super so that any future updates to multisrc would also update MangaDistrict.
Thanks.

Reply:
would it be possible to remove those chapters based on the url slug? so you can simply filter the result of super.chapterListParse